src/plugins/agentDesktop/agentdesktop.ts

```diff
@@ -13,38 +18,100 @@ class AgentDesktopPlugin {
 
     }
     onHostCreate() {
-        let me:any = this;
+        let me: any = this;
         let cwInstance = me.hostInstance;
 
-        cwInstance.on("jwtGrantSuccess", (response:any) => {
+        cwInstance.on("jwtGrantSuccess", (response: any) => {
             if (!this.agentDesktopInfo) {
                 me.onInit();
-                this.config=me.extend(me,response)
+                this.config = me.extend(me, response)
                 //me.AgentDesktop(response.userInfo.userId, response);
                 this.agentDesktopInfo = new AgentDesktopPluginScript(this.config);
             }
         });
 
     }
     onInit() {
-        // let me = this;
+        console.log(Notification.permission, "message")
+        if (Notification.permission === "default") {
+            Notification.requestPermission();
+        }
+        let me: any = this;
         this.appendVideoAudioElemnts()
-        // this.koreCoBrowse = me.koreCoBrowseInit({});
-        // this.rrweb = me.rrwebInit({});
+
+   
+
+        // typing event style setting for anget and bot typing
+        me.hostInstance.on('onWSMessage', (event: any) => {
+            console.log("websocketEvents", event)
+            if (event.messageData.type === "bot_response" && event.messageData?.message[0]?.component?.payload?.payload?.text) {
+                if (Notification.permission !== "granted") return;
+                // this.closeDesktopNotification();
+                let _self: any = this;
+                const text = event.messageData?.message[0]?.component?.payload?.payload?.text;
+                const image = "";
+                const title = "Agent is available now";
+                const options = {
+                    body: text,
+                    icon: "",
+                    vibrate: [200, 100, 200],
+                    sound: "https://uat.kore.ai/agents/sounds/Basic%20Alert%20High.mp3",
+                    badge: "",
+                    image: '',
+                    requireInteraction: false,
+                    onclick: (event: any) => {
+                        event.target.close();
+                    }
```

Comment:
@santhoshvishalve-Kore Make sure the have only desktop notifications related changes in the PR.